storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -545,7 +557,9 @@ public void cleanup() throws IOException {
                         timeIndex.deleteIfExists();
                         return null;
                     }, () -> {
-                        txnIndex.deleteIfExists();
+                        if (txnIndexOpt.isPresent()) {
+                            txnIndexOpt.get().deleteIfExists();
+                        }
```

Comment:
you can use txnIndexOpt.ifPresent()

Reply:
Same here. It would become something like this.
```
  txnIndexOpt.ifPresent(transactionIndex -> {
      try {
          transactionIndex.deleteIfExists();
      } catch (IOException e) {
          throw new RuntimeException(e);
      }
  });
```